### PR TITLE
Adds a debug button that will automatically render an entry to the chemistry wiki page

### PR DESCRIPTION
**Pull request by Thalpy**
## About The Pull Request

Hand writing entries to the wiki is pain. Now you just press a button and it's done for you! (Though- existing entries still need their description retained as it uses the ingame one on output).

It is written for ease of understanding above all rather than optimised for speed (since it should never be used in game).

## Why It's Good For The Game

Editing the wiki is the last thing someone wants to do after coding, testing, tweaking and PR writing. (not to be mean to the wiki writers! It's just a little exhausting. Now it's done for you!

## Changelog
:cl:
qol: adds a wiki render button for reagent entries that will automatically write a table entry for you.
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
This is epic, if this is merged I will make this work for food too.

**Comment by Time-Green** [comment]
Extremely based, could this be the future of quantitative wikipedia pages?

**Comment by Angustmeta** [comment]
I was considering pinging you to learn this power.

**Comment by Thalpy** [comment]
> 
> 
> I was considering pinging you to learn this power.
> 
> Only minor concern is capitalization inconsistencies of chem names compared to current wiki format, but that really is minor compared to the benefits.

If you fix it in the code base, then it should be fixed for the render too! The only caveat is that the wiki tends to have more detailed and better descriptions than in game, so I'm not sure if we should try to have the same description across the wiki and code. 

Oh and feel free to ping me over any wiki concerns, I'm happy to talk about them! I know I'm making a lot of changes so I don't want to step on any toes.

**Comment by Thalpy** [comment]
> 
> 
> This is epic, if this is merged I will make this work for food too.

If you need help, let me know!

**Comment by Cyberboss** [comment]
> 
> 
> > I was considering pinging you to learn this power.
> > Only minor concern is capitalization inconsistencies of chem names compared to current wiki format, but that really is minor compared to the benefits.
> 
> If you fix it in the code base, then it should be fixed for the render too! The only caveat is that the wiki tends to have more detailed and better descriptions than in game, so I'm not sure if we should try to have the same description across the wiki and code.
> 
> Oh and feel free to ping me over any wiki concerns, I'm happy to talk about them! I know I'm making a lot of changes so I don't want to step on any toes.

The code should be the ultimate source of truth in this case.

**Comment by Angustmeta** [comment]
> The code should be the ultimate source of truth in this case.

I think most of the current code chem descriptions are inferior and would be quite upset to see them replace the detailed ones currently on wiki.

**Comment by Thalpy** [comment]
A lot of the descriptions on the wiki are far more indepth, some of them are collapsible too. The recipe lookup in game also has a hotlink directly to the reagent on the wiki as well so I don't think it would work as a copy and paste over.

**Comment by Cyberboss** [comment]
What I'm saying is we should probably have detailed wiki descriptions in game that generate the wiki. The issue with sourcing them on the wiki is one change can throw everything out of balance really fast.

File: code/modules/reagents/chemistry/chem_wiki_render.dm [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+			continue
+
+		for(var/datum/chemical_reaction/reaction as anything in reactions)
+			var/single_parse = generate_chemwiki_line(reagent, reaction)
+			text2file(single_parse, "[GLOB.log_directory]/chem_parse.txt")
+	text2file("|}", "[GLOB.log_directory]/chem_parse.txt") //Cap off the table
+	to_chat(usr, "Done! Saved file to (wherever your root folder is, i.e. where the DME is)/[GLOB.log_directory]/chem_parse.txt OR use the Get Current Logs verb under the Admin tab. (if you click Open, and it does nothing, that's because you've not set a .txt default program! Try downloading it instead, and use that file to set a default program! Have a nice day!")
+
+
+//Generate the big list of reagent based reactions.
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/chem_wiki_render.dm:44]:
```suggestion
/// Generate the big list of reagent based reactions.
```

Doc comment.

File: code/modules/reagents/chemistry/chem_wiki_render.dm [hunk 2]
```diff
@@ -0,0 +1,199 @@
+//Generates a wikitable txt file for use with the wiki - does not support productless reactions at the moment
+/client/proc/generate_wikichem_list()
+	set category = "Debug"
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/chem_wiki_render.dm:3]: Indent with tabs, not spaces.

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#tabs-not-spaces

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/chem_wiki_render.dm (hunk 2 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/chem_wiki_render.dm:3]: Are you sure it's spaces? VSC seems to think it's tabs and doesn't detect any changes when I use conver to indentation.

Comment (Mothblocks) [reply code/modules/reagents/chemistry/chem_wiki_render.dm:3]: I think it's just GitHub being shitty, actually.

**Pull request merged**